suite-common/wallet-utils/src/accountUtils.ts

```diff
@@ -1153,3 +1155,59 @@ export const isTokenMatchesSearch = (token: TokenInfo, search: string) => {
         token.policyId?.toLowerCase().includes(search)
     );
 };
+
+export const isDebugOnlyAccountType = (
+    accountType: AccountType,
+    symbol?: NetworkSymbol,
+): boolean => {
+    if (!symbol) return false;
+
+    const network = (networks as Networks)?.[symbol];
+
+    if (!network) return false;
+
+    const accountTypeInfo = network.accountTypes[accountType];
+
+    return !!accountTypeInfo?.isDebugOnlyAccountType;
+};
+
+type FilterRecieveAccountsProps = {
+    accounts: Account[];
+    deviceState: string | undefined;
+    receiveNetwork?: string;
+    isDebug: boolean;
+    receiveNetworks: Network[];
+};
+
+export const filterRecieveAccounts = ({
+    accounts,
+    deviceState,
+    receiveNetwork,
+    isDebug,
+    receiveNetworks,
+}: FilterRecieveAccountsProps): Account[] => {
+    const isSameDevice = (account: Account) => account.deviceState === deviceState;
+    const isSameNetwork = (account: Account) => account.symbol === receiveNetwork;
+    const isDebugAndIsAccountDebugOnly = (account: Account) =>
+        isDebugOnlyAccountType(account.accountType, account.symbol) && isDebug;
+    const isNotDebugOnlyAccount = (account: Account) =>
+        !isDebugOnlyAccountType(account.accountType, account.symbol);
+    const isNotEmptyAccount = (account: Account) => !account.empty;
+    const isVisibleAccount = (account: Account) => account.visible;
+    const isFirstNormalAccount = (account: Account) =>
+        account.accountType === 'normal' && account.index === 0;
+
+    if (receiveNetworks.length > 0) {
```

Comment:
I think you do not need this condition. Filter of `[]` will return `[]`

Reply:
This check was there initially for another reason I think, but not sure why, it was there before I refactored the checks

<img width="575" alt="image" src="https://github.com/user-attachments/assets/be1be708-7a57-45f7-9232-5409aad76e60">

Reply:
I think it's a check wether the network is supported by suite at all and if not, it would be impossible to buy. So I would let it be there to not break anything that was written before me

Reply:
Because before the method returned `undefined`. Now you return `[]`

Reply:
imho if it should return `[]` then the condition really does not need to be there.

Reply:
Oh right, I see now